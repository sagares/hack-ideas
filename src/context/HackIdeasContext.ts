import { createContext } from "react";
import { IdeasArray } from "../HackIdea.types";

type Employee = {
  empId: string;
};

type HackIdeasContextType = {
  currentUser: Employee;
  logout: () => void;
  setCurrentUser: (employee: Employee) => void;
  ideas: IdeasArray;
  setIdeas: (ideas: IdeasArray) => void;
};

const HackIdeasContext = createContext<Partial<HackIdeasContextType>>({});

export { HackIdeasContext };
