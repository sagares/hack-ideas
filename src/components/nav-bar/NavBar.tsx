import React, { useContext } from "react";
import HackIdeasContext from "../../context";
import { useSessionStorage } from "../../hooks";

import "./nav-bar.less";

const NavBar = () => {
  const { logout } = useContext(HackIdeasContext);
  const [userInSession, setUserInSession] = useSessionStorage("employee", {});

  const handleLogout = () => {
    logout();
    setUserInSession(null);
  };

  return (
    <nav>
      <div className="nav-wrapper">
        <a className="brand-logo">{`{ Hack Ideas }`}</a>
        <ul className="right hide-on-med-and-down">
          <li>
            <a className="logout-btn" onClick={handleLogout}>
              Logout
            </a>
          </li>
        </ul>
      </div>
    </nav>
  );
};

export default NavBar;
