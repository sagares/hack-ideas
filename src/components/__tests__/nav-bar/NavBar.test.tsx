/**
 * @jest-environment jsdom
 */
import React from "react";
import { fireEvent, render } from "@testing-library/react";
import HackIdeasContext from "../../../context";
import NavBar from "../../nav-bar";

describe("Nav Bar Component", () => {
  let element, component, logout;

  beforeEach(() => {
    logout = jest.fn().mockImplementation(() => {});
    element = document.createElement("div");

    component = render(
      <HackIdeasContext.Provider
        value={{
          logout,
        }}
      >
        <NavBar />
      </HackIdeasContext.Provider>,
      {
        container: document.body.appendChild(element),
      }
    );
  });

  afterEach(() => {
    document.body.removeChild(element);
  });

  test("renders", () => {
    expect(component.container).toMatchSnapshot();
  });

  test("logouts on clicking logout button", () => {
    const logoutBtn = element.querySelector(".logout-btn");
    fireEvent.click(logoutBtn);
    expect(logout).toHaveBeenCalled();
  });
});
