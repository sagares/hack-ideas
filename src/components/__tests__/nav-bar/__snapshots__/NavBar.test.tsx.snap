// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Bar Component renders 1`] = `
<div>
  <nav>
    <div
      class="nav-wrapper"
    >
      <a
        class="brand-logo"
      >
        { Hack Ideas }
      </a>
      <ul
        class="right hide-on-med-and-down"
      >
        <li>
          <a
            class="logout-btn"
          >
            Logout
          </a>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;
